plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestHiveS3MinioQueries.java

```diff
@@ -26,29 +26,37 @@
 import static com.google.common.base.Verify.verify;
 import static io.trino.testing.TestingNames.randomNameSuffix;
 import static io.trino.testing.containers.Minio.MINIO_ACCESS_KEY;
+import static io.trino.testing.containers.Minio.MINIO_REGION;
 import static io.trino.testing.containers.Minio.MINIO_SECRET_KEY;
+import static java.lang.String.format;
 import static java.nio.charset.StandardCharsets.UTF_8;
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class TestHiveS3MinioQueries
         extends AbstractTestQueryFramework
 {
-    private Minio minio;
+    private HiveMinioDataLake hiveMinioDataLake;
+    private String bucketName;
 
     @Override
     protected QueryRunner createQueryRunner()
             throws Exception
     {
-        minio = closeAfterClass(Minio.builder().build());
-        minio.start();
+        this.bucketName = "test-hive-minio-queries-" + randomNameSuffix();
+        this.hiveMinioDataLake = closeAfterClass(new HiveMinioDataLake(bucketName));
+        this.hiveMinioDataLake.start();
 
         return HiveQueryRunner.builder()
                 .setHiveProperties(ImmutableMap.<String, String>builder()
-                        .put("hive.metastore.disable-location-checks", "true")
-                        .put("hive.s3.aws-access-key", MINIO_ACCESS_KEY)
-                        .put("hive.s3.aws-secret-key", MINIO_SECRET_KEY)
-                        .put("hive.s3.endpoint", minio.getMinioAddress())
-                        .put("hive.s3.path-style-access", "true")
+                        .put("fs.hadoop.enabled", "false")
+                        .put("fs.native-s3.enabled", "true")
+                        .put("hive.metastore", "thrift")
+                        .put("hive.metastore.uri", hiveMinioDataLake.getHiveHadoop().getHiveMetastoreEndpoint().toString())
+                        .put("s3.aws-access-key", MINIO_ACCESS_KEY)
+                        .put("s3.aws-secret-key", MINIO_SECRET_KEY)
+                        .put("s3.region", MINIO_REGION)
+                        .put("s3.endpoint", hiveMinioDataLake.getMinio().getMinioAddress())
+                        .put("s3.path-style-access", "true")
```

Comment:
We could remove these properties by using `S3HiveQueryRunner.builder(hiveMinioDataLake)`.